app/services/search/postgres/chat_channel.rb

```diff
@@ -0,0 +1,40 @@
+module Search
+  module Postgres
+    class ChatChannel
+      MEMBERSHIP_STATUSES = %w[active joining_request].freeze
+
+      # NOTE: this is "kinda" working. To test this load the console and compare:
+      # 1 - `Search::ChatChannelMembership.search_documents(params: {channel_text: NAME_OF_ORG})`
+      # => vs `Search::Postgres::ChatChannel.search_documents(NAME_OF_ORG)`
+      # 2 - `Search::ChatChannelMembership.search_documents(params: {channel_text: NAME_OF_USER})`
+      # => vs `Search::Postgres::ChatChannel.search_documents(NAME_OF_USER)`
+      # You will see that the number of returned items is the same for both, the difference is that the
+      # ES based index contains a lot more info which is stored at indexing time related to membership
+      # but we don't have that info inside the DB so we need to use a serializer to get it
+      # Also, the JS layer kinda expects a ChatChannelMembership row, not a ChatChannel
```

Comment:
This. I've not written any tests for this, just played with it in the console. Still unsure if this is the best strategy as it does quite a few tricks

Reply:
I think "best strategy" is outside the scope of the spike. It seems to be a viable strategy though, improving on that would be part of the RFC anyway.